required/latex-lab/latex-lab-math.dtx

```diff
@@ -1463,6 +1473,164 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \subsection{Token-by-token inline grabbing}
+%
+% Grabbing inline math token-by-token is more involved. The mechanism here
+% is essentially a simplified version of that originally seen in
+% \pkg{collcell} and refined in \pkg{siunitx}. We make use of the fact that
+% in math mode spaces are ignored, so we have to deal with only \texttt{N}-type
+% tokens and groups. Furthermore, there is no need to look inside groups, so
+% the only special cases are a small selection of \texttt{N}-type tokens.
+%
+% \begin{variable}{\l_@@_grabbed_tl}
+%   For collection of the material piecewise.
+%    \begin{macrocode}
+\tl_new:N \l_@@_grabbed_tl
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{variable}{\l_@@_grab_env_int}
+%   Needed to count up the number of nested environments encountered.
+%    \begin{macrocode}
+\int_new:N \l_@@_grab_env_int
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{macro}{\@@_grab_dollar_loop:}
+% \begin{macro}{\@@_grab_loop:}
+%   The lead-off here establishes a group: we need that as we will have to
+%   be careful in the way \tn{cr} is handled and ensure this is only
```

Comment:
\cr and & ?

Reply:
`&` doesn't need any handling as we never pick it up: TeX inserts the end-of-template tokens, we read the ending `$` and the loop stops

Reply:
treu

Reply:
It's true that a primitive `&` isn't seen but there are other things beside `\\ ` that hide an end of cell, notably packages using active `&` (tikz matrix of nodes being a common example). In many ways tabular and `\\ ` is just a common case of the underlying problem of end of math being behind a macro, which can occur anywwhere not just in tables. Probably should not try to handle the general case here but instead at some point (not in this PR) provide packages with a way of declaring inline math constructs that do safely grab the content.

Reply:
well, maybe the case of active & actually warrants detection and handling though. Couldn't that been done by simply expanding it if enountered?

Reply:
I guess active-`&` can be picked up by `\futurelet` comparison with the appropriate definition - but I think we want to see if this is really an issue before adjusting.

Reply:
don't we already know that it will be needed for, say, tikz stuff as david mentioned? But I'm ok with merging without it at this point in time and adjust later